pkg/util/setutil/set.go

```diff
@@ -20,6 +20,10 @@ func NewSetFromSlice[A any, B cmp.Ordered](slice []A, f func(a A) B) Set[B] {
 	return out
 }
 
+func NewStringSetFromSlice(slice []string) Set[string] {
+	return NewSetFromSlice(slice, func(s string) string { return s })
+}
+
```

Comment:
This function seems redundant. The caller can use `NewSetFromSlice(strings, Identity)`.